src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerMessageHandler.java

```diff
@@ -0,0 +1,78 @@
+     *       Or Status messages from a running backup...<br>
+     *   </li>
+     *   <li><b>"Universal" error logging</b><br>
+     *       Your error-message handler can contain the logic for handling all error logging.
+     *   </li>
+     *   <li><b>"Universal" error handling</b><br>
+     *       Error-handling logic can be placed in your error-message handler, instead of being repeated throughout your application.
+     *   </li>
+     *   <li><b>Remapping of error-message severity</b>, based on application requirements<br>
+     *       Your error-message handler can contain logic for recognizing specific error messages, and downgrading or upgrading 
+     *       their severity based on application considerations rather than the severity rating of the server. 
+     *       For example, during a cleanup operation that deletes old rows, you might want to downgrade the severity of a 
+     *       message that a row does not exist. However, you may want to upgrade the severity in other circumstances.
+     *   </li>
+     * </ul>
+     * 
+     * Example code:
+     * <pre>
+     *  public ISQLServerMessage messageHandler(ISQLServerMessage srvErrorOrWarning)
+     *  {
+     *      ISQLServerMessage retObj = srvErrorOrWarning;
+     *
+     *      if (srvErrorOrWarning.isErrorMessage()) {
+     *
+     *          // Downgrade: 2601 -- Cannot insert duplicate key row...
+     *          if (2601 == srvErrorOrWarning.getErrorNumber()) {
+     *              retObj = srvErrorOrWarning.getSQLServerMessage().toSQLServerInfoMessage();
+     *          }
+     *
+     *          // Discard: 3701 -- Cannot drop the table ...
+     *          if (3701 == srvErrorOrWarning.getErrorNumber()) {
+     *              retObj = null;
+     *          }
+     *      }
+     *
+     *      return retObj;
+     *  }
+    
+     * </pre>
+     * 
+     * @param srvErrorOrWarning
+     * @return 
+     * <ul>
+     *   <li><b>unchanged</b> same object as passed in.<br>
+     *       The JDBC driver will works as if no message hander was installed<br>
+     *       Possibly used for logging functionality<br>
+     *   </li>
+     *   <li><b>null</b><br>
+     *       The JDBC driver will <i>discard</i> this message. No SQLException will be thrown
+     *   </li>
+     *   <li><b>SQLServerInfoMessage</b> object<br>
+     *       Create a "SQL warning" from a input database error, and return it. 
+     *       This results in the warning being added to the warning-message chain.
+     *   </li>
+     *   <li><b>SQLServerError</b> object<br>
+     *       If the originating message is a SQL warning (SQLServerInfoMessage object), messageHandler can evaluate 
+     *       the SQL warning as urgent and create and return a SQL exception (SQLServerError object)
+     *       to be thrown once control is returned to the JDBC Driver.
+     *   </li>
+     * </ul>
+     */
+    ISQLServerMessage messageHandler(ISQLServerMessage srvErrorOrWarning);

```

Comment:
nitpick: since this is a public API, can you please use a full spelling instead
```suggestion
    ISQLServerMessage messageHandler(ISQLServerMessage serverErrorOrWarning);

```

Reply:
done